en/concept/geometry_files.md

```diff
@@ -0,0 +1,80 @@
+# Geometry Files
+
+- Associated dev guide issue: https://github.com/PX4/PX4-Devguide/issues/349
+- Geometry files are found here: https://github.com/PX4/PX4-Autopilot/tree/master/src/lib/mixer/MultirotorMixer/geometries
+- They are TOML files
+
+@[youtube](https://youtu.be/orvng_11ngQ?t=1080) - Control Allocation - 18 minute in.
+
+[Slide 14 has info](https://static.sched.com/hosted_files/px4developersummitvirtual2020/1b/PX4%20Developer%20Summit%202020%20-%20Overview%20of%20multicopter%20control%20from%20sensors%20to%20motors.pdf)
+
+<!-- stuff just copied from the slide
+In a nutshell
+Desired forces
+and torques (m)
+?
+Actuator
+outputs (u)
+Control allocation
+matrix
+Actuator
+effectiveness
+matrix (B)
+Inverse
+Control
+allocation
+Matrix (P=B
++
+Moore-Penrose )
+-->
+
+
+## How to add a new geometry
+
+1. Create new TOML geometry file in [/src/lib/mixer/MultirotorMixer/geometries](https://github.com/PX4/PX4-Autopilot/tree/master/src/lib/mixer/MultirotorMixer/geometries).
+   For example "foo.toml".
+   with a new key (e.g.: key = "4fo") and add to **CMakeLists.txt** <!-- which one? -->
+
+2. In ROMFS/px4fmu_common/Create new mixer file **mixers/foo.main.mix** with a line containing the new key:
+   ```
+   R 4fo 10000 10000 10000 0  
+   ```
+   <!-- There was a comment that the numbers are not needed? Why are they needed in quad then -->
+3. Set the new mixer in **init.d/airframes/myconfig**
+   ```
+   set MIXER foo
+   ```
+   
+   
+## Fragment from some files
+
+Looking at the toml files they all have this first. So I am assuming file description follwed by info. 
+- Is key random?
+- Where is the description used?
+
+```
+# Generic Quadcopter in H configuration
+
+[info]
```

Comment:
What are the things that can be in each section? Just these listed? What are axis, ct, cm, position, direction, and how would YOU go about calculating them?

Reply:
Those fields are allowed: https://github.com/PX4/PX4-Autopilot/blob/58b40fbbb921ad306f97da040acd7f141496484c/src/lib/mixer/MultirotorMixer/geometries/tools/px_generate_mixers.py#L103
- **name**: name of the rotor, used to make the geometry file more readable
- **position**: a vector describing the location of the rotor in the body frame (Forward-Right-Down) centered on the CG. Can be in meters or other units, only the ratio between the distances is really important since the mixer is normalized.
- **axis**: a vector in the in the body frame (Forward-Right-Down) describing the direction of the thrust produced by the rotor. For example `[1.0, 0.0, -1.0]` means that the rotor produces upward and forward thrust equally (i.e.: angle of 45 degrees).
- **direction**: specifies the direction of rotation of a rotor, `CW` (clockwise) or `CCW` (counter clockwise)
- **Ct**: non-dimensional thrust coefficient. For example, a rotor with a `Ct` of 2.0 produces 2 times the thrust of a rotor with a `Ct` of 1.0
- **Cm**: the non-dimensional drag torque coefficient relates to the axial torque produced by a spinning propeller. This needs to be set relative to `Ct` if some torque can be produced by drag and thrust together. If set to 0, the mixer will assume that the rotor does not produce any axial torque.